server/server.go

```diff
@@ -42,72 +55,362 @@ func HandleWebsocketConnection(serverHTTPResponse http.ResponseWriter, clientHTT
 	if websocketConnectionErr != nil {
 		// If there was an error while upgrading the HTTP connection to a WebSocket connection return a relevant error
 		http.Error(serverHTTPResponse, "Unable to upgrade HTTP connection to a WebSocket connection", 400)
+		// Close the websocket connection
+		websocketConnection.Close()
 		return
 	}
-
 	// Require successful authentication before allowing a client to send a message
-	_, websocketConnectionStateErr := requireAuth(websocketConnection)
-	if websocketConnectionStateErr != nil {
+	authenticatedIdentityPublicKeyHexClient, websocketConnectionrequestAuthErr := requestAuth(websocketConnection)
+	if websocketConnectionrequestAuthErr != nil {
 		// If the authentication failed, terminate the websocket connection to the client
-		log.Println("Authentication Failed:", websocketConnectionStateErr)
-		websocketConnection.WriteMessage(websocket.BinaryMessage, []byte(websocketConnectionStateErr.Error()))
+		log.Println("Authentication Failed:", websocketConnectionrequestAuthErr)
+		websocketConnection.WriteMessage(websocket.BinaryMessage, []byte(websocketConnectionrequestAuthErr.Error()))
 		log.Println("Terminating websocket connection to client.")
+		// Close the websocket connection
+		websocketConnection.Close()
 		return
 	}
+	// Check if there are pending messages to be delivered when the client comes back online
+	if messagesToBeDelivered, ok := multiUserChatMessage[authenticatedIdentityPublicKeyHexClient]; ok {
+		deliverMessages(websocketConnection, messagesToBeDelivered)
+	} // if messagesToBeDelivered
 
-	// Notify the client that authentication succeeded
-	websocketConnection.WriteMessage(websocket.BinaryMessage, []byte("Authentication Succeeded"))
-
+	websocketConnectionProcessMessageErr := processMessage(websocketConnection, authenticatedIdentityPublicKeyHexClient)
```

Comment:
we don't use `websocketConnectionProcessMessageErr`. We should at least log it and / or send it back to the client.

Reply:
Noted.